### ISSUE TITLE
[Clock Module][1.12.2]

### ISSUE THREAD

**Issue by jmc529** [SquidDev-CC/ccjam-2016#71]
CC: Tweaked 1.1 Plethora 1.1.8
Clock module's method, getDay() only returns 0. Am I missing something or is something amiss?

**Comment by jmc529** [comment]
To avoid creating new tickets, in case I'm just being dumb: I also think the shoot method for skeletons isn't working properly. It's returning Java Exception Thrown wrong number of arguments. To replicate: "lua> modules.shoot([number])" number being b/w 0.1 and 1.0 

The other issue I'm having is with the mkII. it's contents: note block, clock, daylight sensor, and laser. using "lua> modules.wrap([direction])" then "lua> modules." to query the methods available (for testing purposes) results in only the first item actually being registered.  (Exp If I placed the note block in first, the only specialized methods available to me would be playSound() and playNote().

Also it might be worth mentioning that the speaker added by CC: tweaked doesn't have a getDocs() and I can't seem to find any documentation on it. I want to play the sounds! *animated tears* All this said, thank you for this mod and your work! I like it a lot.

Also, since I'm here... Might I suggest a method to access mobs interfaces without taking yours off. Shift-use perhaps? or is this already implemented?

**Comment by SquidDev** [comment]
>  I also think the shoot method for skeletons isn't working properly. It's returning `Java Exception Thrown wrong number of arguments`.

Yep, this is a bug - I'm a bit of a plonker and broke something in the 1.10 to 1.11 update. I'm not sure how I didn't notice but the fix should be in the next release.

> Using `lua> modules.wrap([direction])` then `lua> modules.` to query the methods available results in only the first item actually being registered.

Out of interest, what does `modules.listModules()` produce? Does running `modules = peripheral.wrap([direction])` again make the other methods show up?

> Also it might be worth mentioning that the speaker added by CC: tweaked doesn't have a `getDocs()` and I can't seem to find any documentation on it.

There isn't much documentation for the new features of CC/CC: Tweaked. I've made a start at writing some, but haven't got round to speakers yet. The methods are pretty much the same as Plethora's note block methods:

 - `.playNote(instrument: string[, volume:number[, pitch:number]])`: Play a noteblock note. `instrument` is one of `harp`, `bass`, `snare`, `hat`, `flute`, `bell`, `chime`, and `guitar`.
 - `.playSound(name: string[, volume:number[, pitch:number]])`: `name` is a resource name, like that used in `/playsound`.

> Also, since I'm here... Might I suggest a method to access mobs interfaces without taking yours off. Shift-use perhaps? or is this already implemented?

Historically right clicking would prefer the entity over yourself. However it appears something changed in the 1.12 update breaking this functionality. I'll have a prod and see what needs changing :)

---

I'm really sorry about all the issues you've encountered, but thank you very much for reporting them!

**Comment by jmc529** [comment]
@SquidDev Regarding the mkII, interestingly enough this only happens when you change a module on the mkII (not mkI). It seems like it's not logging the addition of new modules correctly. I have to relog, break the block, or clear the board for it to update the peripheral. Also, this causes the internal server to shutdown on relog, "A fatal error has occurred, this connection is terminated"

After removing or adding a  module to the set without clearing the updating using the methods above, .listModules() returns, "lua:1: A module has changed". Using .wrap() again has no affect.

I tested it out in a world with just  CC and Plethora, and I found this issue to be nonexistent. So I assume it is some mod clashing. 
Mod List:
-snip-
